simd/src/Kokkos_SIMD_Scalar.hpp

```diff
@@ -86,29 +126,74 @@ class basic_simd_mask<T, simd_abi::scalar> {
+      basic_simd_mask const& lhs, basic_simd_mask const& rhs) noexcept {
+    return basic_simd_mask(static_cast<bool>(lhs) < static_cast<bool>(rhs));
+  }
 };
 
+template <class T>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr bool all_of(
+    basic_simd_mask<T, Kokkos::Experimental::simd_abi::scalar> const&
+        a) noexcept {
+  return static_cast<bool>(
+      a == basic_simd_mask<T, Kokkos::Experimental::simd_abi::scalar>(true));
+}
+
+template <class T>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr bool any_of(
+    basic_simd_mask<T, Kokkos::Experimental::simd_abi::scalar> const&
+        a) noexcept {
+  return static_cast<bool>(
+      a != basic_simd_mask<T, Kokkos::Experimental::simd_abi::scalar>(false));
+}
+
+template <class T>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr bool none_of(
+    basic_simd_mask<T, Kokkos::Experimental::simd_abi::scalar> const&
+        a) noexcept {
+  return static_cast<bool>(
+      a == basic_simd_mask<T, Kokkos::Experimental::simd_abi::scalar>(false));
+}
+
 template <class T>
 class basic_simd<T, simd_abi::scalar> {
   T m_value;
 
  public:
-  using value_type                       = T;
-  using abi_type                         = simd_abi::scalar;
-  using mask_type                        = basic_simd_mask<T, abi_type>;
-  KOKKOS_DEFAULTED_FUNCTION basic_simd() = default;
-  KOKKOS_DEFAULTED_FUNCTION basic_simd(basic_simd const&)            = default;
-  KOKKOS_DEFAULTED_FUNCTION basic_simd(basic_simd&&)                 = default;
-  KOKKOS_DEFAULTED_FUNCTION basic_simd& operator=(basic_simd const&) = default;
-  KOKKOS_DEFAULTED_FUNCTION basic_simd& operator=(basic_simd&&)      = default;
+  using value_type = T;
+  using abi_type   = simd_abi::scalar;
+  using mask_type  = basic_simd_mask<T, abi_type>;
+
   KOKKOS_FORCEINLINE_FUNCTION static constexpr std::size_t size() { return 1; }
+
+  KOKKOS_DEFAULTED_FUNCTION constexpr basic_simd() noexcept         = default;
+  KOKKOS_DEFAULTED_FUNCTION constexpr basic_simd(basic_simd const&) = default;
+  KOKKOS_DEFAULTED_FUNCTION constexpr basic_simd(basic_simd&&)      = default;
+  KOKKOS_DEFAULTED_FUNCTION constexpr basic_simd& operator=(basic_simd const&) =
+      default;
+  KOKKOS_DEFAULTED_FUNCTION constexpr basic_simd& operator=(basic_simd&&) =
+      default;
   template <class U, std::enable_if_t<std::is_convertible_v<U, value_type>,
                                       bool> = false>
-  KOKKOS_FORCEINLINE_FUNCTION basic_simd(U&& value) : m_value(value) {}
+  KOKKOS_FORCEINLINE_FUNCTION constexpr basic_simd(U&& value) noexcept
+      : m_value(value) {}
   template <class U, std::enable_if_t<std::is_convertible_v<U, value_type>,
                                       bool> = false>
-  KOKKOS_FORCEINLINE_FUNCTION explicit basic_simd(
-      basic_simd<U, abi_type> const& other)
+  KOKKOS_FORCEINLINE_FUNCTION constexpr explicit basic_simd(
+      basic_simd<U, abi_type> const& other) noexcept
```

Comment:
As for the simd_mask it looks like this also in the standard enables conversion from different ABIs.

Reply:
Yes, I will also add this when updating #6466